### PR TITLE
redis: fix several stubtest warnings

### PR DESCRIPTION
**Pull request by Akuli**
Stubtest command used:

```
MYPYPATH=stubs/redis python3 -m mypy.stubtest --concise redis
```

<details>
<summary>Fixed these 45 stubtest warnings:</summary>

```
redis.AuthenticationWrongNumberOfArgsError is not present in stub
redis.ChildDeadlockedError is not present in stub
redis.Sentinel is not present in stub
redis.SentinelConnectionPool is not present in stub
redis.SentinelManagedConnection is not present in stub
redis.SentinelManagedSSLConnection is not present in stub
redis.client.CaseInsensitiveDict is not present in stub
redis.client.Monitor.command_re is not present in stub
redis.client.Monitor.monitor_re is not present in stub
redis.client.Pipeline.client_list is inconsistent, stub does not have argument "_type"
redis.client.Pipeline.client_list is inconsistent, stub does not have argument "client_id"
redis.client.Pipeline.discard is not present in stub
redis.client.Pipeline.flushall is inconsistent, stub does not have argument "asynchronous"
redis.client.Pipeline.flushdb is inconsistent, stub does not have argument "asynchronous"
redis.client.Pipeline.hscan is inconsistent, runtime argument "match" has a default value of type None, which is incompatible with stub argument type builtins.str
redis.client.PubSubWorkerThread is not present in stub
redis.client.Redis.flushall is inconsistent, stub does not have argument "asynchronous"
redis.client.Redis.flushdb is inconsistent, stub does not have argument "asynchronous"
redis.commands.CoreCommands.append is not present in stub
redis.commands.CoreCommands.bitcount is not present in stub
redis.commands.CoreCommands.bitop is not present in stub
redis.commands.CoreCommands.bitpos is not present in stub
redis.commands.CoreCommands.flushall is not present in stub
redis.commands.CoreCommands.flushdb is not present in stub
redis.commands.core.CoreCommands.append is not present in stub
redis.commands.core.CoreCommands.bitcount is not present in stub
redis.commands.core.CoreCommands.bitop is not present in stub
redis.commands.core.CoreCommands.bitpos is not present in stub
redis.commands.core.CoreCommands.flushall is not present in stub
redis.commands.core.CoreCommands.flushdb is not present in stub
redis.connection.HIREDIS_SUPPORTS_BYTE_BUFFER is not present at runtime
redis.connection.HIREDIS_SUPPORTS_CALLABLE_ERRORS is not present at runtime
redis.connection.HIREDIS_USE_BYTE_BUFFER is not present at runtime
redis.connection.hiredis_version is not present at runtime
redis.connection.msg is not present at runtime
redis.connection.parse_url is not present in stub
redis.exceptions.ModuleError is not present in stub
redis.lock.Lock.LUA_EXTEND_SCRIPT is not present in stub
redis.lock.Lock.LUA_REACQUIRE_SCRIPT is not present in stub
redis.lock.Lock.LUA_RELEASE_SCRIPT is not present in stub
redis.lock.Lock.lua_extend is not present in stub
redis.lock.Lock.lua_reacquire is not present in stub
redis.lock.Lock.lua_release is not present in stub
redis.sentinel.Sentinel.execute_command is not present in stub
redis.sentinel.SentinelManagedSSLConnection is not present in stub
```

</details>

393 warnings remain. Most of them are "... is not present in stub" warnings, which should be easy to fix (by either moving methods to base classes such as `CoreCommands`, or by adding stubgen-generated methods).

### CODE REVIEW

**Comment by Akuli** [comment]
Should I split this into smaller parts so that other maintainers are more excited to review it?

**Comment by JelleZijlstra** [comment]
I can review it, but might need a bit more time. (If anyone else gets here before me, don't wait for my review though.)

File: stubs/redis/redis/client.pyi
Comment (JelleZijlstra) [inline stubs/redis/redis/client.pyi:675]:
```suggestion
    def hscan(self, name: _Key, cursor: int = ..., match: str | None = ..., count: int | None = ...) -> Pipeline[_StrType]: ...  # type: ignore[override]
```
It defaults to None

**Review by JelleZijlstra — APPROVED** [review]
Looks good apart from one comment

File: stubs/redis/@tests/stubtest_allowlist.txt
Comment (hauntsaninja) [inline stubs/redis/@tests/stubtest_allowlist.txt:1]:
```suggestion
```

**Pull request merged**
